Docs/architecture/decisions/002-deprecate-response-fallback.md

```diff
@@ -0,0 +1,32 @@
+# 2. Deprecate the response fallback on the RestConfigurableProducer class
+
+Date: 2020-05-08
+
+## Status
+
+Accepted
+
+## Context
+
+Currently the RestConfigurableProducer will pass the payload received from the target system to the serializer. If this one fails, it will catch the exception, fallback to the original response, **and continue the execution**.
+
+This was done to allow certain APIs to return a non-deserializable response while still being able to continue the execution, either **by ignoring the payload** (i.e.: when the response code is enough) or to map it to something else.
+
+This makes the mapper vulnerable, as it has no guarantee that what it will expect is something **that can be mapped**.
+
+Due to the fallback is present by default, one endpoint **could randomly fail** if the target system decides to send back a response in a format that is not expected by this bundle.
```

Comment:
```suggestion
Due to the fallback is present by default, one endpoint **could randomly fail** if the target system decides to send back a response in a format that is not expected by the application.
```

Reply:
Isn't framework a bundle? You cant run framework on its own. Why do you want to change it to application?

Reply:
Because it depends on how the producers are configured, and this is not configured in the framework itself